### DISCUSSION TITLE
Async job failed (Rust LSP keeps crashing)

### DISCUSSION THREAD

**Discussion by Ha5hBr0wn** [helix-editor/helix#8367] [Troubleshooting] [answered]
I am trying to use helix to write Rust code on my MacBook Pro (M1 Max chip, Ventura OS). 

I installed rust via the suggestion on the [website](https://www.rust-lang.org/tools/install)

And I installed rust-analyzer and helix via brew. 

```
rust-analyzer --version
rust-analyzer 0.0.0 (99718d0c8 2023-07-22)
```

```
hx --version
helix 23.05 (7f5940be)
```

```
rustup --version
rustup 1.26.0 (5af9b9484 2023-04-05)
info: This is the version for the rustup toolchain manager, not the rustc compiler.
info: The currently active `rustc` version is `rustc 1.71.1 (eb26296b5 2023-08-03)`
```

```
rustc --version
rustc 1.71.1 (eb26296b5 2023-08-03)
```

The issue is whenever I save the code such that it would experience some compilation issues I get `Async job failed (some number)`. And then my editor also has trouble quitting, such that when I enter `:q` the editor just hangs for a long time and sometimes I have to just exit the whole terminal and lose work. 

Any suggestions on how to fix this?

**Comment by the-mikedavis** [comment]
What does the log file say when running in verbose mode (`hx -v`)? https://github.com/helix-editor/helix/wiki/FAQ#access-the-log-file

**Comment by Ha5hBr0wn** [comment, marked as answer]
I have actually fixed the issue by uninstalling helix and rust analyzer with brew. And then installed helix in the following way (mostly just following the docs, but putting it here in case it helps someone else)

```
git clone https://github.com/helix-editor/helix
cd helix
git checkout 23.05 <---- this I added to go to the most recent release
cargo install --path helix-term --locked
ln -s $PWD/runtime ~/.config/helix/runtime <---- config.toml and languages.toml go under ~/.config/helix
```

I then followed the links in the docs to the lsp setup page [here](https://github.com/helix-editor/helix/wiki/How-to-install-the-default-language-servers) and just followed what was there. 

`rustup component add rust-analyzer`

In languages.toml:
```
[[language]]
name = "rust"

# If this doesn't work, you might be on a older release version which handles this differently.
# Try using [language.config.check] instead of [language-server.rust-analyzer.config.check]
[language-server.rust-analyzer.config.check]
command = "clippy"
```

Everything seems to be working great now.
